src/main/java/seedu/address/ui/EntryCard.java

```diff
@@ -40,10 +42,17 @@ public EntryCard(Entry entry) {
         super(FXML);
         this.entry = entry;
         entryName.setText(entry.getEntryName().name);
-        startDate.setText("From: "
-                + entry.getStartDate().format(DateTimeFormatter.ofPattern("E, dd MMM yyyy h:mm a")));
-        endDate.setText("To: "
-                + entry.getEndDate().format(DateTimeFormatter.ofPattern("E, dd MMM yyyy h:mm a")));
+
+        if (entry.haveDifferentDates()) {
```

Comment:
Thanks! In this case you can also delete the Label startTime and endTime and perhaps we can change "Date" to "Due date" to be clearer?